src/jdk.jlink/share/classes/jdk/tools/jlink/internal/Platform.java

```diff
@@ -24,48 +24,104 @@
  */
 package jdk.tools.jlink.internal;
 
-import jdk.tools.jlink.plugin.ResourcePoolModule;
-
 import java.util.Locale;
 
 /**
  * Supported platforms
  */
-public enum Platform {
-    WINDOWS,
-    LINUX,
-    MACOS,
-    AIX,
-    UNKNOWN;
+public record Platform(OperatingSystem os, Architecture arch) {
 
-    /**
-     * Returns the {@code Platform} derived from the target platform
-     * in the {@code ModuleTarget} attribute.
+    public enum OperatingSystem {
+        WINDOWS,
+        LINUX,
+        MACOS,
+        AIX,
+        UNKNOWN;
+    }
+
+    public enum Architecture {
+        X86,
+        x64,
+        ARM,
+        AARCH64,
+        UNKNOWN;
+    }
+
+    public static final Platform UNKNOWN = new Platform(OperatingSystem.UNKNOWN, Architecture.UNKNOWN);
+
+    /*
+     * Returns the {@code Platform} based on the platformString of the form <operating system>-<arch>.
      */
-    public static Platform toPlatform(String targetPlatform) {
+    public static Platform parseTargetPlatform(String platformString) {
```

Comment:
This method can be renamed back to `parsePlatform` (an earlier version).    My suggestion to name this method as `parseTargetPlatform` no longer apply in this version.

Reply:
Changed the name back to `parsePlatform`.